lib/matplotlib/figure.py

```diff
@@ -2430,6 +2436,14 @@ def axes(self):
 
     get_axes = axes.fget
 
+    def _get_cached_renderer(self):
```

Comment:
This is the main new (private) method.  Users should generally not need to fetch the renderer going forward...

Reply:
I find the naming a bit confusing. I'd suggest just `_get_renderer(self)`, the caching should be an implementation detail. The `if self.cachedRenderer is not None` seems like it should be implemented in the `self.canvas.get_renderer()` method (and indeed, there is some logic to re-use the Agg renderer for this) https://github.com/matplotlib/matplotlib/blob/7b2d828fe0702cbdee7f80714f0a9779c4ac68cc/lib/matplotlib/backends/backend_agg.py#L415-L425